raiden/transfer/state.py

```diff
@@ -1274,9 +1278,12 @@ def to_dict(self) -> Dict[str, Any]:
     @classmethod
     def from_dict(cls, data: Dict[str, Any]) -> 'UnlockProofState':
         restored = cls(
-            merkle_proof=map_list(serialization.deserialize_bytes, data['merkle_proof']),
+            merkle_proof=map_list(
```

Comment:
Are these `cast`s necessary? It looks like `deserialize_bytes` should be type instead of adding the cast here.

Reply:
> It looks like deserialize_bytes should be type instead of adding the cast here.

I don't understand what you are trying to say here.

Reply:
He means that the typecast here would be unnecessary if `deserialize_bytes` wasn't declared as
```python
def deserialize_bytes(data: str) -> bytes:
```
but as
```python
def deserialize_bytes(data: str) -> Keccak256:
```

And since that is a good thing anyway, it's better to do it now and remove the typecheck here.

Reply:
Yeah but deserialize bytes is a generic function to deserialize string into any type of bytes and is/can also be used for things other than `Keccak256`.

Reply:
Ah right, sorry for missing the point. Mypy supports overloading type signatures, but that only helps when the input type differs.

Reply:
I also think the casts overcomplicate the readability of the codebase. I think the only way to workaround this is golang-style, where we create a function for each type. Each of those functions would call the cast rather than us repeating the casts here.

Reply:
As discussed in the office, I removed casts in favour of some specialized functions here: https://github.com/raiden-network/raiden/pull/3800/commits/b6d9d5ce4c12b988e6c409d28ff09c53e625d795